src/engine/source/server/src/endpoints/unixDatagram.cpp

```diff
@@ -0,0 +1,264 @@
+#include <server/endpoints/unixDatagram.hpp>
+
+#include <cstring>      // Unix  socket datagram bind
+#include <fcntl.h>      // Unix socket datagram bind
+#include <sys/socket.h> // Unix socket datagram bind
+#include <sys/un.h>     // Unix socket datagram bind
+#include <unistd.h>     // Unix socket datagram bind
+
+#include <logging/logging.hpp>
+#include <uvw.hpp>
+
+namespace
+{
+constexpr unsigned int MAX_MSG_SIZE {65536 + 512}; ///< Maximum message size (TODO: I think this should be 65507)
+} // namespace
+
+namespace engineserver::endpoint
+{
+UnixDatagram::UnixDatagram(const std::string& address,
+                           const std::function<void(std::string&&)>& callback,
+                           const std::size_t taskQueueSize)
+    : Endpoint(address, taskQueueSize)
+    , m_callback(callback)
+    , m_handle(nullptr)
+    , m_bufferSize(-1)
+{
+    if (address.empty())
+    {
+        throw std::runtime_error("Address must not be empty");
+    }
+
+    if (address.length() >= sizeof(sockaddr_un::sun_path))
+    {
+        auto msg = fmt::format("Path '{}' too long, maximum length is {} ", address, sizeof(sockaddr_un::sun_path));
+        throw std::runtime_error(msg);
+    }
+
+    if (m_address[0] != '/')
+    {
+        throw std::runtime_error("Address must start with '/'");
+    }
+
+    if (!callback)
+    {
+        throw std::runtime_error("Callback must be set");
+    }
+}
+
+UnixDatagram::~UnixDatagram()
+{
+    if (isBound())
+    {
+        // Close
+        m_handle->close();
+        m_handle = nullptr;
+        unlink(m_address.c_str());
+
+    }
+}
+
+void UnixDatagram::bind(std::shared_ptr<uvw::Loop> loop)
+{
+    if (isBound())
+    {
+        throw std::runtime_error("Endpoint already bound");
+    }
+
+    m_loop = loop;
+    m_handle = m_loop->resource<uvw::UDPHandle>();
+
+    // Listen for incoming data
+    m_handle->on<uvw::UDPDataEvent>(
+        [this](const uvw::UDPDataEvent& event, uvw::UDPHandle& handle)
+        {
+            // Get the data
+            auto data = std::string {event.data.get(), event.length};
```

Comment:
can we make it const?

Reply:
No, since it would not be possible to apply movement semantics.